readyplayerme/rpm-unreal-next-gen-sdk
# RpmActor now supports setting/mapping of skeleton and animation blueprints

## [TICKETID](https://ready-player-me.atlassian.net/browse/TICKETID)

## Description

![image](https://github.com/user-attachments/assets/1aff3259-cb71-442a-bef0-c6501fa2074b)


- RpmActor now has an `AnimationConfig` that is exposed to blueprints (and C++ functions). This can be used to set which skeleton and animation Blueprint to use. If you don't need to support multiple "styles" then this is all you need
![image](https://github.com/user-attachments/assets/666e94ee-0d25-4857-ac21-343f27ea4888)

- to support multiple styles (multiple base modes) there is also a new property exposed to blueprints called `AnimationConfigsByBaseModelId`. This can be used to map different styles/base models to their appropriate skeletons and animations blueprints. If nothing is set then animations won't run for that style/basemodel. 
![image](https://github.com/user-attachments/assets/ee95a39e-13d0-4eac-867e-6822e0484305)
- Category icons and Asset panels are now created dynamically based on the categories fetched from RPM API's. As long as you are signed in and have authentication. 

-   various things were refactored to make this work
      - Separation of LoadCharacter vs LoadAsset functions on RPMActor (to better handle loading a full character vs 1 single asset) 




## Changes

#### Added

-   List your additions and new features here.

#### Updated

-   List your updates and changes here.

#### Removed

-   List what is removed here.



## How to Test

-   Add steps to locally test these changes



## Checklist


## QA Testing

-  Test both samples 
         - test that the UI works automatically without any manual setup of category names (also if the category/type names match the default it should auto load default category icons )
 - Check all previous functionality works
           - developer window login (both with custom rpm dev account and demo)
           - importing/loading of base models (both with custom account and demo)
 - test that animations work (simplest way is to use the Phoenix-Unreal repo
       - alternatively, you can try yourself in a fresh project but it will require some manual setup (I will start working on a doc for this) 

## Details

-   If more details are necessary to support the description with images and videos add them here.
